File: LeetCode/49 - Group Anagrams.rb
# @param {String[]} strs
# @return {String[][]}
def group_anagrams(strs)
    hash = {}
    ans = []
    strs.each { |val| 
        hash[val.chars.sort.join] ||= []
        hash[val.chars.sort.join] << val
    }
    hash.each { |key, value| ans << value }
    ans
end

File: LeetCode/49. Group Anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        map<string, vector<string> > myMap;
        vector<vector<string> > ans;

        for (auto str : strs) {
            string temp = str;
            sort(str.begin(), str.end());

            myMap[str].push_back(temp);
        }

        for (auto it : myMap) {
            ans.push_back(it.second);
        }

        return ans;
    }
};
